src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,216 @@
+        }
+
+        // Output the current type info of the target node and the conversion type that the target node is going to be casted by
+        private void GetTypeInfo(SemanticModel semanticModel, SyntaxNode root, SyntaxNode targetNode, CancellationToken cancellationToken, out ITypeSymbol nodeType, out ITypeSymbol conversionType)
+        {
+            var nodeInfo = semanticModel.GetTypeInfo(targetNode, cancellationToken);
+            nodeType = nodeInfo.Type;
+            conversionType = nodeInfo.ConvertedType;
+
+            // CS1503 
+            var textSpan = targetNode.GetLocation().SourceSpan;
+            TryGetNode(root, textSpan, SyntaxKind.Argument, targetNode, out var argument);
+            if (argument != null)
+            {
+                conversionType = null;
+                var argumentListNode = argument.Parent;
+                var invocationNode = argumentListNode.Parent;
+                var symbolInfo = semanticModel.GetSymbolInfo(invocationNode, cancellationToken);
+                var candidcateSymbols = symbolInfo.CandidateSymbols;
+
+                var argumentList = (ArgumentListSyntax)argumentListNode;
+
+                // Find all valid candidates and extract potential conversion types 
+                var potentialConversionTypes = new List<ITypeSymbol> {};
+                foreach (var candidcateSymbol in candidcateSymbols)
+                {
+                    var methodSymbol = (IMethodSymbol)candidcateSymbol;
+                    var parameterList = methodSymbol.Parameters;
+                    if (parameterList.Length != argumentList.Arguments.Count)
+                    {
+                        continue;
+                    }
+
+                    // Test if all parameters are convertible, otherwise it is not the perfect match function
+                    var allValid = true;
+                    for (var i = 0; i < parameterList.Length; i++)
+                    {
+                        var argType = semanticModel.GetTypeInfo(argumentList.Arguments[i].Expression, cancellationToken);
+                        if (!IsTypeConvertible(argType.Type, parameterList[i].Type))
+                        {
+                            allValid = false;
+                            break;
+                        }
+                    }
+
+                    if (allValid)
+                    {
+                        var targetArgumentIndex = argumentList.Arguments.IndexOf((ArgumentSyntax)argument);
+                        var correspondingParameter = parameterList[targetArgumentIndex];
+                        potentialConversionTypes.Add(correspondingParameter.Type);
+                    }
+                }
+
+                // If there is no exact solution, then don't provide suggestions
+                if (potentialConversionTypes.Count != 1)
+                {
+                    return;
+                }
+
+                conversionType = potentialConversionTypes[0];
```

Comment:
looks like we could avoid the allocation of potentialConversionTypes since you only care about the 0/1 cases.